test/topology_custom/test_mv_backlog.py

```diff
@@ -0,0 +1,112 @@
+        if requested_labels:
+            # we know metric begins with name{ and ends with } - the labels
+            # are what we have between those
+            got_labels = metric[len(name)+1:-1].split(',')
+            # Check that every one of the requested labels is in got_labels:
+            for k, v in requested_labels.items():
+                if not f'{k}="{v}"' in got_labels:
+                    # No match for requested label, skip this metric (python
+                    # doesn't have "continue 2" so let's just set val to 0...
+                    val = 0
+                    break
+        total += float(val)
+    return total
+
+def get_replicas(cql, key):
+    return cql.cluster.metadata.get_replicas("ks", Int32Type.serialize(key, cql.cluster.protocol_version))
+
+async def enable_on_servers(manager, servers, injection, oneshot=False):
+    errs = [manager.api.enable_injection(s.ip_addr, injection, oneshot) for s in servers]
+    return await asyncio.gather(*errs)
+async def disable_on_servers(manager, servers, injection):
+    errs = [manager.api.disable_injection(s.ip_addr, injection) for s in servers]
+    return await asyncio.gather(*errs)
+
+def get_remote_key(cql, local_node):
+    i = 0
+    while local_node == get_replicas(cql, i)[0]:
+        i = i + 1
+    return i
+
+# This test reproduces issue #18542
+# In the test, we create a table and perform a write to it a couple of times
+# Each time, we check that a view update backlog on some shard increased
+# due to the write.
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_view_backlog_increased_after_write(manager: ManagerClient) -> None:
+    node_count = 2
+    # Use a higher smp to make it more likely that the writes go to a different shard than the coordinator.
+    servers = await manager.servers_add(node_count, cmdline=['--smp', '5'])
+    cql = manager.get_cql()
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 1}}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    key = int(time.time())
+    logger.info(f"Base table key: {key}")
+    local_node = get_replicas(cql, key)[0]
+    # Only remote updates hold on to memory, so make the update remote
+    remote_key = get_remote_key(cql, local_node)
+    for v in [1000, 4000, 16000, 64000, 256000]:
+        # Make sure the view update backlog is still high (view udpate hasn't finished) when the replica returns
+        await enable_on_servers(manager, servers, "never_finish_remote_view_updates")
+
+        await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {remote_key}, '{v*'a'}')")
```

Comment:
It is deliberate that we aren't using prepared statements here, right? AFAIK we want to trigger a non-shard-aware query. If so, then I think this should be explained in a comment.

Reply:
Hmm, `key = int(time.time())` and `remote_key = get_remote_key(cql, local_node)`, but __key__ is the partition key, not remote_key. Was this intended?

Reply:
@piodul yes - I think he aimed for the partition key on the **view**, which is c (not "key") to be remote_key.

Reply:
Oh, right, of course :) Thanks for the explanation.

Perhaps the column names could be changed to something more explicit (`key` -> `base_key`, `c` -> `mv_key`) so that obtuse people like me can understand it more quickly.

Reply:
Right, to lower the confusion here, I modified the names